src/lib_ccx/dvb_subtitle_decoder.c

```diff
@@ -1695,53 +1747,8 @@ int dvbsub_decode(struct encoder_ctx *enc_ctx, struct lib_cc_decode *dec_ctx, co
 						segment_length);
 				break;
 			case DVBSUB_DISPLAY_SEGMENT: //when we get a display segment, we save the current page
-				if (enc_ctx->write_previous) //this condition is used for the first subtitle - write_previous will be 0 first so we don't encode a non-existing previous sub
-				{
-					sub->prev->end_time = (dec_ctx->timing->current_pts - dec_ctx->timing->min_pts) / (MPEG_CLOCK_FREQ / 1000); //we set the end time of the previous sub the current pts
-					encode_sub(enc_ctx->prev, sub->prev); //we encode it
-					enc_ctx->srt_counter = enc_ctx->prev->srt_counter; //for dvb subs we need to update the current srt counter because we always encode the previous subtitle (and the counter is increased for the previous context)
-					enc_ctx->prev_start = enc_ctx->prev->prev_start;
-					sub->prev->got_output = 0;
-					if (enc_ctx->write_format == CCX_OF_WEBVTT) {	// we already wrote header, but since we encoded last sub, we must prevent multiple headers in future
-						enc_ctx->wrote_webvtt_header = 1;
-					}
-				}
-				/* copy previous encoder context*/
-				free_encoder_context(enc_ctx->prev);
-				enc_ctx->prev = NULL;  
-				enc_ctx->prev = copy_encoder_context(enc_ctx); 
-				/* copy previous decoder context */
-				free_decoder_context(dec_ctx->prev);
-				dec_ctx->prev = NULL;
-				dec_ctx->prev = copy_decoder_context(dec_ctx);
-				freep(&dec_ctx->prev->private_data);
-				dec_ctx->prev->private_data = malloc(sizeof(struct DVBSubContext));
-				memcpy(dec_ctx->prev->private_data, dec_ctx->private_data, sizeof(struct DVBSubContext));
-				/* copy previous subtitle */
-				free_subtitle(sub->prev);
-				sub->prev = NULL;
-				sub->prev = copy_subtitle(sub);
-				sub->prev->start_time = (dec_ctx->timing->current_pts - dec_ctx->timing->min_pts) / (MPEG_CLOCK_FREQ / 1000); //we set the start time of the previous sub the current pts
-				
-				write_dvb_sub(dec_ctx->prev, sub->prev); //we write the current dvb sub to update decoder context
-				enc_ctx->write_previous = 1; //we update our boolean value so next time the program reaches this block of code, it encodes the previous sub
+				dvbsub_handle_display_segment(enc_ctx, dec_ctx, sub);
 				got_segment |= 16;
-#ifdef ENABLE_OCR
```

Comment:
@harrynull  Was this no longer needed?

Reply:
@saurabhshri I move the whole thing into a function named dvbsub_handle_display_segment to reuse the code.

Reply:
@harrynull I viewed the diff on phone and I missed this part. Sorry :)